lib/task_helpers/metrics.rb

```diff
@@ -0,0 +1,49 @@
+module TaskHelpers
+  class Metrics
+    class << self
+      def per_tenant
+        column_names = %w[
+          tenant
+          portfolio_count
+          product_count
+          portfolio_share_count
+        ].freeze
+        data = [column_names.collect(&:titleize)]
+
+        Tenant.order(:external_tenant).each do |current_tenant|
+          ActsAsTenant.with_tenant(current_tenant) do
+            next if tenant_empty?
+
+            data << column_names.collect { |column_name| send(column_name) }
+          end
+        end
+
+        data
+      end
+
+      def tenant
+        ActsAsTenant.current_tenant.external_tenant
+      end
+
+      def portfolio_count
+        Portfolio.count
+      end
+
+      def portfolio_share_count
+        Portfolio.count do |portfolio|
+          portfolio.metadata['statistics']['shared_groups'] > 0
+        end
```

Comment:
@gmcculloug Correct me if I'm wrong here, but is this actually doing what we want? If something has multiple shared groups, this will only count it once, right? Does John want to know the total amount of shared groups, or does he just want to know if it has been shared to at least one group?

Reply:
He's looking for two things in this area:
```
# of Portfolios Shared
# of groups per Portfolios Shared per Organization
```

Since this PR is focused on the Tenant level reporting we can only report on the first item, since the second one requires a per-Portfolio breakdown.  I would say we are good here.